testng 7.1.0 java.lang.ClassNotFoundException: com.google.inject.Stage

### TestNG Version
TestNG 7.1.0

### Expected behavior


### Actual behavior

the testng-remote integration test failed
https://travis-ci.org/testng-team/testng-remote/jobs/481901771

```
[RemoteTestNG] revisions:
	git.commit.id=32c8977
	git.branch=32c8977911e409c8c8c2d4bce973e94d2acf5ca7
	git.build.version=1.4.1-SNAPSHOT
[RemoteTestNG] loaded class org.testng.internal.Version at file:/home/travis/.groovy/grapes/org.testng/testng/jars/testng-7.1.0.jar
[RemoteTestNG] detected TestNG version 7.1.0
Caught: java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: com/google/inject/Stage
java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: com/google/inject/Stage
	at org.testng.internal.Configuration.<init>(Configuration.java:33)
	at org.testng.TestNG.init(TestNG.java:216)
	at org.testng.TestNG.<init>(TestNG.java:200)
	at org.testng.remote.AbstractRemoteTestNG.<init>(AbstractRemoteTestNG.java:17)
	at org.testng.remote.support.RemoteTestNG6_12.<init>(RemoteTestNG6_12.java:18)
	at org.testng.remote.support.RemoteTestNGFactory6_12.createRemoteTestNG(RemoteTestNGFactory6_12.java:16)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:67)
	at org.testng.remote.RemoteTestNG$main.call(Unknown Source)
	at TestNGTest.run(TestNGTest.groovy:49)
Caused by: java.lang.NoClassDefFoundError: com/google/inject/Stage
	... 9 more
Caused by: java.lang.ClassNotFoundException: com.google.inject.Stage
	... 9 more
[BaseMessageSender] Stopped receiver
```

### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans


### Test case sample
> Please, share the test case (as small as possible) which shows the issue
https://github.com/testng-team/testng-remote/blob/master/remote-test/src/test/groovy/Tester.groovy

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,4 +1,5 @@
 Current
+Fixed: GITHUB-2223: testng 7.1.0 java.lang.ClassNotFoundException: com.google.inject.Stage (Krishnan Mahadevan)
 Fixed:GITHUB-217: Configure TestNG to fail when all tests are skipped (Krishnan Mahadevan)
 Fixed: GITHUB-2255: Ensure test method parameters are visible in BeforeMethod config method (Krishnan Mahadevan)
 Fixed: GITHUB-2251: NullPointerException at test with timeOut (Krishnan Mahadevan)

--- a/src/main/java/org/testng/internal/Configuration.java
+++ b/src/main/java/org/testng/internal/Configuration.java
@@ -1,6 +1,5 @@
 package org.testng.internal;
 
-import com.google.inject.Guice;
 import java.util.List;
 import java.util.Map;
 import org.testng.IConfigurable;
@@ -30,7 +29,7 @@ public class Configuration implements IConfiguration {
   private boolean alwaysRunListeners = true;
   private IExecutorFactory m_executorFactory = new DefaultThreadPoolExecutorFactory();
 
-  private IInjectorFactory injectorFactory = Guice::createInjector;
+  private IInjectorFactory injectorFactory = new GuiceBackedInjectorFactory();
 
   public Configuration() {
     init(new JDK15AnnotationFinder(new DefaultAnnotationTransformer()));

--- a/src/main/java/org/testng/internal/GuiceBackedInjectorFactory.java
+++ b/src/main/java/org/testng/internal/GuiceBackedInjectorFactory.java
@@ -1,0 +1,15 @@
+package org.testng.internal;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Stage;
+import org.testng.IInjectorFactory;
+
+class GuiceBackedInjectorFactory implements IInjectorFactory {
+
+  @Override
+  public Injector getInjector(Stage stage, Module... modules) {
+    return Guice.createInjector(stage, modules);
+  }
+}

